firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java

```diff
@@ -1029,6 +1031,46 @@ public void testMultipleUpdatesWhileOffline() {
     assertEquals(asList(map("foo", "zzyzx", "bar", "2")), querySnapshotToValues(snapshot2));
   }
 
+  @Test
+  public void resumingQueryShouldRemoveDeletedDocumentsIndicatedByExistenceFilter() {
+    Map<String, Map<String, Object>> testDocs = new LinkedHashMap<>();
+    for (int i = 1; i <= 100; i++) {
+      testDocs.put("doc" + i, map("key", i));
+    }
+
+    // Setup firestore with disabled persistence and populate a collection with testDocs.
+    FirebaseFirestore firestore = testFirestore();
+    firestore.setFirestoreSettings(
```

Comment:
I don't think that disabling persistence is necessary. It doesn't matter if the cache is populated with the documents that were written by `writeAllDocs()`. Or am I missing something?

Reply:
Without disabling persistence, it is always returning 100 docs after deleting the 50 docs. 
And in web SDK, it is the same result and we skipped the test if persistence is enabled as well.

Reply:
Something doesn't seem right then. If persistence is disabled then the resume token will be garbage collected as soon as the first call to `collection.get()` completes. Then, later on, when it calls `collection.get()` the 2nd time, the client will not specify a resume token in the request since there isn't one in persistence. For this test to be valid, we _need_ persistence enabled in order to have a resume token specified in the 2nd request, which triggers the bloom filter logic.

I took a closer look at the corresponding PR in the web sdk, https://github.com/firebase/firebase-js-sdk/pull/7045, and now I remember that I forgot to ask you about this back when I reviewed it. I believe the test in the web sdk is invalid as well: that integration test _only_ makes sense when persistence is _enabled_.

That's really weird that the 2nd call to `collection.get()` would return the original 100 documents. This needs to be investigated. Or maybe my brain isn't functioning properly and you can let me know!

Reply:
Discussed offline.